gk/fib.c

```diff
@@ -1261,28 +1266,35 @@ init_grantor_fib_locked(
 	gt_fib->action = GK_FWD_GRANTOR;
 	gt_fib->u.grantor.set = rte_malloc("gk_fib.grantor.set",
 		sizeof(*(gt_fib->u.grantor.set)) +
-			sizeof(*(gt_fib->u.grantor.set->entries)), 0);
+			num_addrs * sizeof(*(gt_fib->u.grantor.set->entries)),
+		0);
 	if (gt_fib->u.grantor.set == NULL) {
 		GK_LOG(ERR, "Could not allocate set of Grantor entries\n");
-		return NULL;
+		goto init_fib;
 	}
 	gt_fib->u.grantor.set->proto = ip_prefix->addr.proto;
-	gt_fib->u.grantor.set->num_entries = 1;
-	gt_fib->u.grantor.set->entries[0].gt_addr = *gt_addr;
-	gt_fib->u.grantor.set->entries[0].eth_cache = eth_cache;
+	gt_fib->u.grantor.set->num_entries = num_addrs;
```

Comment:
Bug: `num_addrs` can be larger than what filed `num_entries` can accommodate. We need to verify that `num_addrs` is not too large.